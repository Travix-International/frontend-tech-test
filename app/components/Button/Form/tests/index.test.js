import React from 'react';
import { create } from 'react-test-renderer';

import ButtonForm from '../index';

const initial = {};
const renderComponent = props => {
  const finalProps = { ...initial, ...props };

  return (
    <ButtonForm {...finalProps}>
      Test
    </ButtonForm>
  );
};

describe('<ButtonForm />', () => {
  it('should render component', () => {
    const tree = create(renderComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component when is secondary', () => {
    const props = {
      secondary: true
    };
    const tree = create(renderComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component when is loading', () => {
    const props = {
      loading: true
    };
    const tree = create(renderComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
