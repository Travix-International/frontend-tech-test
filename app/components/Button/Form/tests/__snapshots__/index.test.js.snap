// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonForm /> should render component 1`] = `
<div
  className="Wrapper__Wrapper-dXHlXX ggGEtw"
>
  <button
    className="StyledButton__StyledButton-iQSVZX fjLvae"
    onClick={undefined}
  >
    Test
  </button>
</div>
`;

exports[`<ButtonForm /> should render component when is loading 1`] = `
<div
  className="Wrapper__Wrapper-dXHlXX ggGEtw"
>
  <button
    className="StyledButton__StyledButton-iQSVZX fjLvae"
    onClick={undefined}
  >
    <div
      className="Wrapper__Wrapper-cZxMuO fNDNpp"
      height="1em"
      width="1em"
    >
      <div
        className="Circle__CirclePrimitive-doUHun cEgePr"
      />
      <div
        className="Circle__CirclePrimitive-iczIjT gCBYdH"
      />
      <div
        className="Circle__CirclePrimitive-cmIcdf hWqTMY"
      />
      <div
        className="Circle__CirclePrimitive-jOiEYJ gvkDDz"
      />
      <div
        className="Circle__CirclePrimitive-eAcCGr fQYIRr"
      />
      <div
        className="Circle__CirclePrimitive-fQPULp ijToqU"
      />
      <div
        className="Circle__CirclePrimitive-egCDAQ bQEvSu"
      />
      <div
        className="Circle__CirclePrimitive-kwRWka kpvgpv"
      />
      <div
        className="Circle__CirclePrimitive-jnKZUF kElwxJ"
      />
      <div
        className="Circle__CirclePrimitive-jSRngu htSGkR"
      />
      <div
        className="Circle__CirclePrimitive-gbjHSq fBGPCV"
      />
      <div
        className="Circle__CirclePrimitive-blqIGF bnGgtd"
      />
    </div>
  </button>
</div>
`;

exports[`<ButtonForm /> should render component when is secondary 1`] = `
<div
  className="Wrapper__Wrapper-dXHlXX ggGEtw"
>
  <button
    className="StyledButton__StyledButton-iQSVZX eueAcV"
    onClick={undefined}
  >
    Test
  </button>
</div>
`;
